### PR TITLE
consensus sharding sync and cross shard msg send

### PR DESCRIPTION
**Pull request by xiemylogos**
1、fix cross shard msg send
2、fix shard consensus cycle change
3、optimize shard call tx in p2p send
4、save some cross shard msg to db

### CODE REVIEW

File: consensus/vbft/node_utils.go [hunk 1]
```diff
@@ -497,9 +497,19 @@ func (self *Server) SendCrossShardMsgToAll(height uint32) {
 		if err != nil {
 			if err != com.ErrNotFound {
 				log.Errorf("SendCrossShardMsgToAll getshardmsghash err:%s", err)
+				continue
```
Comment (Honglei-Cong) [inline consensus/vbft/node_utils.go]: continue? or return?

File: core/chainmgr/chainmgr.go [hunk 2]
```diff
@@ -77,6 +76,7 @@ type ChainManager struct {
 	lock       sync.RWMutex
 	shards     map[common.ShardID]*ShardInfo
 	mainLedger *ledger.Ledger
+	ledger     *ledger.Ledger
```
Comment (Honglei-Cong) [inline core/chainmgr/chainmgr.go]: It's better to remove both mainLedger and ledger from chainmgr.
Getting ledger with ledger.GetShardLedger(...)?

File: consensus/vbft/node_utils.go (hunk 1 above)
Comment (xiemylogos) [reply consensus/vbft/node_utils.go]: maintain data consistency,use return

File: core/chainmgr/chainmgr.go (hunk 2 above)
Comment (xiemylogos) [reply core/chainmgr/chainmgr.go]: ok

**Review by Honglei-Cong — APPROVED** [review]

**Pull request merged**